File: 250820/자주 등장한 top K 숫자/top-k-frequent-elements.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const [n, k] = input[0].split(' ').map(Number);
const arr = input[1].trim().split(' ').map(Number);

const freq = new Map();
for (const v of arr) {
    freq.set(v, (freq.get(v) || 0) + 1);
}

const items = Array.from(freq.entries());


items.sort((a, b) => {
    if (b[1] === a[1]) {
        return b[0] - a[0];
    }
    return b[1] - a[1];
});

const result = items.slice(0, k).map(([num, _]) => num);


console.log(result.join(" "));

File: 250820/자주 등장한 top K 숫자/top-k-frequent-elements.py
n, k = map(int, input().split())
arr = list(map(int, input().split()))

temp = {}

for v in arr:
    if v in temp:
        temp[v] += 1
    else:
        temp[v] = 1

temp2 = [(x, y) for x, y in temp.items()]
temp2.sort(key=lambda x: (-x[1], -x[0]))

result = [num for num, _ in temp2[:k]]
print(" ".join(map(str, result)))